src/appleseed-max-impl/appleseedinteractive/interactivetilecallback.h

```diff
@@ -0,0 +1,70 @@
+
+//
+// This source file is part of appleseed.
+// Visit http://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2017 Sergo Pogosyan, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#pragma once
+
+// appleseed-max headers.
+#include "appleseedrenderer/tilecallback.h"
+
+// appleseed.foundation headers.
+#include "foundation/image/tile.h"
+#include "foundation/platform/types.h"
+#include "foundation/platform/windows.h"    // include before 3ds Max headers
+
+// 3ds Max headers.
+#include <bitmap.h>
+#include <interactiverender.h>
+
+// Standard headers.
+#include <future>
+
+// Forward declarations.
+namespace renderer  { class Frame; }
+class Bitmap;
+class IIRenderMgr;
+
+
+class InteractiveTileCallback
+  : public TileCallback
+{
+  public:
+      InteractiveTileCallback(
+        Bitmap*                         bitmap,
+        IIRenderMgr*                    iimanager,
+        renderer::IRendererController*  render_controller);
+
+    virtual void on_progressive_frame_end(const renderer::Frame* frame) override;
+    static void update_caller(UINT_PTR param_ptr);
```

Comment:
I really don't see why `update_caller()` cannot be made private. What am I missing?

Reply:
That means you can pass pointer to private methods... Interesting..

Reply:
> That means you can pass pointer to private methods... Interesting..

Yep, private members & methods are just a safeguard for programmers, there is absolutely no runtime checks associated with them (it has zero cost).

In C/C++ you can always get, pass or even craft a pointer to a method and then call it.